ui/ozone/platform/x11/x11_display_fetcher_ozone.cc

```diff
@@ -0,0 +1,274 @@
+          crtc(XRRGetCrtcInfo(xdisplay_, resources.get(), output_info->crtc));
+
+      int64_t display_id = -1;
+      if (!display::EDIDParserX11(output_id).GetDisplayId(
+              static_cast<uint8_t>(i), &display_id)) {
+        // It isn't ideal, but if we can't parse the EDID data, fallback on the
+        // display number.
+        display_id = i;
+      }
+
+      gfx::Rect crtc_bounds(crtc->x, crtc->y, crtc->width, crtc->height);
+      display::Display display(display_id, crtc_bounds);
+
+      if (!display::Display::HasForceDeviceScaleFactor()) {
+        display.SetScaleAndBounds(device_scale_factor, crtc_bounds);
+      }
+
+      if (has_work_area) {
+        gfx::Rect intersection_in_pixels = crtc_bounds;
+        if (is_primary_display) {
+          intersection_in_pixels.Intersect(work_area_in_pixels);
+        }
+        // SetScaleAndBounds() above does the conversion from pixels to DIP for
+        // us, but set_work_area does not, so we need to do it here.
+        display.set_work_area(gfx::Rect(
+            gfx::ScaleToFlooredPoint(intersection_in_pixels.origin(),
+                                     1.0f / display.device_scale_factor()),
+            gfx::ScaleToFlooredSize(intersection_in_pixels.size(),
+                                    1.0f / display.device_scale_factor())));
+      }
+
+      switch (crtc->rotation) {
+        case RR_Rotate_0:
+          display.set_rotation(display::Display::ROTATE_0);
+          break;
+        case RR_Rotate_90:
+          display.set_rotation(display::Display::ROTATE_90);
+          break;
+        case RR_Rotate_180:
+          display.set_rotation(display::Display::ROTATE_180);
+          break;
+        case RR_Rotate_270:
+          display.set_rotation(display::Display::ROTATE_270);
+          break;
+      }
+
+      if (is_primary_display)
+        explicit_primary_display_index = display_id;
+
+      auto monitor_iter = output_to_monitor.find(output_id);
+      if (monitor_iter != output_to_monitor.end() && monitor_iter->second == 0)
+        monitor_order_primary_display_index = display_id;
+
+      gfx::ColorSpace color_space;
+      if (!display::Display::HasForceDisplayColorProfile()) {
+        gfx::ICCProfile icc_profile = ui::GetICCProfileForMonitor(
+            monitor_iter == output_to_monitor.end() ? 0 : monitor_iter->second);
+        icc_profile.HistogramDisplay(display_id);
+        color_space = icc_profile.GetColorSpace();
+      } else {
+        color_space = display::Display::GetForcedDisplayColorProfile();
+      }
+
+      displays.push_back(display);
+    }
+  }
+
+  if (explicit_primary_display_index != -1) {
+    primary_display_index = explicit_primary_display_index;
+  } else if (monitor_order_primary_display_index != -1) {
+    primary_display_index = monitor_order_primary_display_index;
+  }
+
+  if (displays.empty()) {
+    observer_->AddDisplay(GetFallbackDisplay(), true);
+    return;
+  }
+
+  for (auto& display : displays) {
```

Comment:
let this method return a vector of displays and make the initial caller to call AddDisplay. Otherwise, the name of the method doesn't correlate with what it does.

Reply:
Done.